tasks/gulp/__tests__/after-build-package.test.js

```diff
@@ -1,112 +1,96 @@
-              // Remove /src prefix from filenames
-              const fileWithoutSrc = file.replace(/^src\//, '')
-
-              // Account for govuk-esm folder
-              if (fileWithoutSrc.split('.').pop() === 'mjs') {
-                const esmFile = fileWithoutSrc.replace('govuk/', 'govuk-esm/')
-                const umdFile = fileWithoutSrc.replace('.mjs', '.js')
-
-                return [umdFile, esmFile]
-              } else {
-                return fileWithoutSrc
-              }
-            })
-            // Allow for additional files that are not in src
-            .concat(filesNotInSrc)
-            .flat()
-            // Sort to make comparison easier
-            .sort()
-        },
-        error => {
-          console.error('Unable to get package files', error)
-        }
-      )
-    }
-
-    // Compare the expected directory listing with the files we expect
-    // to be present
-    return Promise.all([actualPackageFiles(), expectedPackageFiles()])
-      .then(results => {
-        const [actualPackageFiles, expectedPackageFiles] = results
-
-        expect(actualPackageFiles).toEqual(expectedPackageFiles)
-      })
+  it('should contain the expected files', async () => {
+    const filterPatterns = [
+      '!**/.DS_Store',
+      '!**/*.test.*',
+      '!**/__snapshots__/',
+      '!**/__snapshots__/**',
+      `!${configPaths.src}README.md`
+    ]
+
+    // Build array of expected output files
+    const filesExpected = [...listingSource]
+      .flatMap(listingToArray)
+
+      // Only include files matching filter patterns
+      .filter(filterPath(filterPatterns))
+
+      // Replaces source component '*.mjs' with:
+      // - `*.mjs` file copied to `govuk-esm`
+      // - `*.js` file compiled to `govuk`
+      .flatMap(mapPathTo(['**/*.mjs'], ({ dir: requirePath, name }) => {
+        const importPath = normalize(slash(requirePath).replace('src/govuk', 'src/govuk-esm'))
+
+        return [
+          join(importPath, `${name}.mjs`),
+          join(requirePath, `${name}.js`)
+        ]
+      }))
+
+      // Replaces source component '*.yaml' with:
+      // - `fixtures.json` fixtures for tests
+      // - `macro-options.json` component options
+      .flatMap(mapPathTo([`${configPaths.components}**/*.yaml`], ({ dir }) => [
+        join(dir, 'fixtures.json'),
+        join(dir, 'macro-options.json')
+      ]))
+
+      // Files output from 'src' to 'package'
+      .map((file) => normalize(slash(file).replace(/^src\//, 'package/')))
+
+      // Files already present in 'package'
+      .concat(...[
+        normalize('package/govuk-prototype-kit.config.json'),
+        normalize('package/package.json'),
+        normalize('package/README.md')
+      ])
+      .sort()
+
+    // Build array of actual output files
+    const filesActual = [...listingPackage]
+      .flatMap(listingToArray)
+      .sort()
+
+      // Only include files matching filter patterns
+      .filter(filterPath(filterPatterns))
```

Comment:
**issue**: At second read, the listing for the actual files shouldn't exclude the same filterPatterns as the expected files. By doing it in both places, we'll miss if one of the excluded files has been included by error.

Reply:
@romaricpascal This was a great spot, thanks

It was slightly useful at ignoring `.DS_Store` file if you'd opened the destination directories in Finder, but yep we don't want any stray tests or snapshots being wrongly ignored

All filters have now been removed from the "expected" arrays